src/core/request.js

```diff
@@ -103,25 +104,46 @@ export default function request(options) {
      * @returns {Promise} resolves with response, or rejects if something went wrong
      */
     const runRequest = () => {
-
         let tempToken;
 
         /**
          * Fetches a security token and appends it to headers, if required
          * Also saves the retrieved token in a temporary constiable, in case we need to re-enqueue it
          * @returns {Promise<Object>} - resolves with headers object
          */
-        const computeHeaders = () => {
-            const headers = {...options.headers};
-            if (!options.noToken) {
-                return tokenHandler.getToken().then(token => {
-                    tempToken = token;
+        const computeCSRFTokenHeader = () => {
+            if (options.noToken) {
+                return Promise.resolve({});
+            }
+            return tokenHandler.getToken().then(token => {
+                tempToken = token;
 
-                    headers[tokenHeaderName] = token || 'none';
-                    return headers;
-                });
+                return { [tokenHeaderName]: token || 'none' };
+            });
+        };
+
+        /**
+         * Fetches a Bearer token if token handler is provided
+         * @returns {Promise<Object>} promise of Bearer token header
+         */
+        const computeBearerTokenHeader = () => {
+            const { bearerTokenHandler } = options;
+            if (bearerTokenHandler) {
+                return bearerTokenHandler.getToken().then(token => ({
+                    Authorization: `Bearer ${token}`
+                }));
             }
-            return Promise.resolve(headers);
+            return Promise.resolve({});
+        };
+
+        /**
+         * Extends header object with token headers
+         * @returns {Promise<Object>} Promise of headers object
+         */
+        const computeHeaders = () => {
+            return Promise.all([computeCSRFTokenHeader(), computeBearerTokenHeader()]).then(tokenHeaders =>
+                tokenHeaders.reduce((headers, tokenHeader) => ({ ...headers, ...tokenHeader }), options.headers)
```

Comment:
Since the reduce is a bit complex, you could add a clearer explanation to the docblock.